packages/@react-spectrum/menu/src/MenuDialogTrigger.tsx

```diff
@@ -50,6 +51,7 @@ function MenuDialogTrigger<T>(props: ItemProps<T> & {isUnavailable?: boolean, ta
           isMobile ? (
             <Modal state={state} isDismissable>
               {content}
+              <DismissButton onDismiss={state.close} />
```

Comment:
Perhaps we should also add a DismissButton before the content as well to mimic Tray/Popover. That way the user can land on the dismiss button when swiping backwards as well as forward